### DISCUSSION TITLE
Add slot to ui.toggle?

### DISCUSSION THREAD

**Discussion by v479038280** [zauberzeug/nicegui#3003] [Ideas / Feature Requests]
It seems we can't add slot or icon to  each option of toggle button,we can only display toggle button with labels?

**Comment by falkoschindler** [comment]
Hi @v479038280,

I think this is currently not possible. When creating these choice elements like `ui.toggle`, `ui.select` and `ui.radio`, we opted for a very simple options API with either a list of values or a dictionary mapping from value to label. Adding support for additional attributes like icons and slots would be great, but we would need to find a way to extend the API without breaking existing projects too badly. And I fear that the already complex implementation could get out of control if we add this extra layer of variability. But we're open for suggestions. 👍🏻
